chunk/chunk.go

```diff
@@ -305,25 +314,36 @@ func (s *ValidatorStore) WithDeliverCallback(f func(Chunk)) *ValidatorStore {
 // Put overrides Store put method with validators check. For Put to succeed,
 // all provided chunks must be validated with true by one of the validators.
 func (s *ValidatorStore) Put(ctx context.Context, mode ModePut, chs ...Chunk) (exist []bool, err error) {
-	for _, ch := range chs {
-		if !s.validate(ch) {
+	chunkTypes := make([]Type, len(chs))
+	for i, ch := range chs {
+		// register all valid chunk types, fail otherwise
+		valid, chunkType := s.validate(ch)
+		if !valid {
 			return nil, ErrChunkInvalid
 		}
+		chunkTypes[i] = chunkType
 	}
-	return s.Store.Put(ctx, mode, chs...)
+	exist, err = s.Store.Put(ctx, mode, chs...)
+	// if callback is defined, call it for every new, valid, content-addressed chunk
+	if err == nil && s.deliverCallback != nil {
```

Comment:
I would rather keep the happy path less indented. To have here check if `err == nil { return false, err }`, then to check explicitly if deliverCallback is set, and at the end to `return exists, nil`.

Reply:
thanks, agree with this.

i implemented this, although i did it this way:
```golang
exist, err = s.Store.Put...
if err != nil {
    return nil, err
}
if s.deliverCallback...
```

(i think this is what you mean rather than `err -- nil`)